scala/lambdas/postingest-state-change-handler/src/test/scala/uk/gov/nationalarchives/postingeststatechangehandler/LambdaTest.scala

```diff
@@ -0,0 +1,172 @@
+      sqsMessagesRef <- Ref[IO].of[Map[String, List[SQSMessage]]](Map(queue1Url -> Nil))
+      snsMessagesRef <- Ref[IO].of[List[OutputMessage]](Nil)
+      _ <- new Lambda().handler(
+        event,
+        config,
+        Dependencies(createDynamoClient(itemsInTableRef, updateTableReqsRef), createSnsClient(snsMessagesRef), createSqsClient(sqsMessagesRef), () => instant, () => messageId)
+      )
+      itemsRemainingInTable <- itemsInTableRef.get
+      updateTableReqs <- updateTableReqsRef.get
+      sqsMessages <- sqsMessagesRef.get
+      snsMessages <- snsMessagesRef.get
+    } yield UpdatedRefs(itemsRemainingInTable, updateTableReqs, sqsMessages, snsMessages)
+  }
+
+  "handler" should "queue the item and send an SQS message to the correct queue if the event is an 'INSERT' one and NewImage exists" in {
+    val oldDynamoItem = None
+    val newDynamoItem = fullyPostIngestedAsset
+    val event = DynamodbEvent(List(DynamodbStreamRecord(EventName.INSERT, StreamRecord(getPrimaryKey(newDynamoItem).some, oldDynamoItem, newDynamoItem))))
+    val refs = runLambda(Nil, event).unsafeRunSync()
+
+    refs.itemsRemainingInTable.size should equal(0)
+
+    refs.updateTableReqs.size should equal(1)
+    val updateTableReq = refs.updateTableReqs.head
+
+    updateTableReq.tableName should equal("ddbTable")
+    updateTableReq.primaryKeyAndItsValue should equal(Map("assetId" -> fromS(newDynamoItem.assetId.toString)))
+    updateTableReq.attributeNamesAndValuesToUpdate("queue") should equal(Some(fromS("CC")))
+    updateTableReq.attributeNamesAndValuesToUpdate("firstQueuedAt") should equal(Some(fromS("2038-01-19T03:14:07Z")))
+    updateTableReq.attributeNamesAndValuesToUpdate("lastQueuedAt") should equal(Some(fromS("2038-01-19T03:14:07Z")))
+
+    refs.sentSqsMessages.size should equal(1)
+    val (queueUrl, messagesPerQueue) = refs.sentSqsMessages.head
+
+    queueUrl should equal(queue1Url)
+    messagesPerQueue.head.getBody should equal(s"""{"assetId":"${fullyPostIngestedAsset.assetId}","batchId":"batchId","resultAttrName":"result_CC","payload":"input"}""")
+
+    refs.sentSnsMessages.size should equal(1)
+    val sentSnsMessage = refs.sentSnsMessages.head
+
+    sentSnsMessage.properties.executionId should equal("batchId")
+    sentSnsMessage.properties.messageId should equal(UUID.fromString("de20e39a-948e-468d-a89d-338af262e0f5"))
+    sentSnsMessage.properties.parentMessageId should equal(Some(correlationId))
+    sentSnsMessage.properties.timestamp should equal(Instant.parse("2038-01-19T03:14:07Z"))
+    sentSnsMessage.properties.messageType should equal(IngestUpdate)
+
+    sentSnsMessage.parameters.assetId should equal(fullyPostIngestedAsset.assetId)
+    sentSnsMessage.parameters.status should equal(IngestedPreservation)
+  }
+
+  forAll(updateTableScenarios) { (oldImage, newImage, description) =>
+    "handler" should f"delete the item if the event is an 'MODIFY' one, $description" in {
```

Comment:
I think you can just use `s""` here, you don't need `f""`

Reply:
👍 😄 That was a mistake; I think I had a slight case of Python-brain when I did this